src/components/pages/Departments/Departments.tsx

```diff
@@ -1,40 +1,27 @@
-import React, { useEffect, useState, useMemo, useContext } from 'react';
-
-import DepartmentsAPI from '../../../API/Departments';
-import { Department } from '../../../types/departments';
+import React, { useEffect, useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import AddButton from '../../AddButton/AddButton';
 import NotFound from '../NotFound/NotFound';
 import DepartmnetsList from '../../DepartmentsList/DepartmentsList';
 import DepartmentForm from '../../DepartmentForm/DepartmentForm';
-import LoaderContext from '../../../contexts/Loader/LoaderContext';
+import { AppThunkDispatch, rootState } from '../../../types/store/generalStateTypes';
+import { fetchDepartmentsThunk } from '../../../store/actionCreators/departmentsActionCreators';
 
 const Departments: React.FC = () => {
-  const [departments, setDepartments] = useState<Department[]>([]);
-
-  const loaderContext = useContext(LoaderContext);
-
-  const getDepartments = async () => {
-    loaderContext!.toggleLoader(loaderContext!.isLoading);
-
-    try {
-      const res = await DepartmentsAPI.getDepartments();
-
-      setDepartments(res.data.data);
-    } catch (err) {
-      console.error(err);
-    }
+  const departments = useSelector((state: rootState) => state.departments.departments?.data);
+  const errMsg = useSelector((state: rootState) => state.departments.errorMsg);
```

Comment:
selectors should come from different file

Reply:
OK thanks